### DISCUSSION TITLE
2.2.0 bug: type object 'Flask' has no attribute 'try_trigger_before_first_request_functions'

### DISCUSSION THREAD

**Discussion by kdieudonne-adeo** [pallets/flask#4727] [Q&A] [answered]
when running gunicorn with flask 2.2.0 i get this error as soon as i start the program, downgrading to 2.1.2 fix this problem

edit: might have been better placed in issues, didn't see it before posting

**Comment by davidism** [comment, marked as answer]
`before_first_request` is deprecated and will be removed in Flask 2.3. As part of that, some internals were refactored for performance, including `try_trigger_before_first_request_functions` method. What code were you using that was calling this method? This method was not intended as a public API, and will be going away completely in the next feature release, so it's probably better to address whatever is using it.

**Reply by kdieudonne-adeo** [reply]
it seems it's dd-trace that use this, they have an issue already opened on their git-hub about it so hopefully it'll be updated soon
thank you for the help

**Reply by kdieudonne-adeo** [reply]
https://github.com/DataDog/dd-trace-py/issues/4025
(link to the ddtrace issue in case others have the same problem)
